aicsimageio/aics_image.py

```diff
@@ -0,0 +1,418 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from typing import Any, List, Optional, Tuple
+
+import dask.array as da
+import numpy as np
+
+from . import types
+from .dimensions import DEFAULT_DIMENSION_ORDER, Dimensions
+from .readers.reader import Reader
+
+###############################################################################
+
+
+class AICSImage:
+
+    # The order of the readers in this list is important.
+    # Example:
+    # if TiffReader was placed before OmeTiffReader,
+    # we would never hit the OmeTiffReader
+    # SUPPORTED_READERS = [
+    #     ArrayLikeReader,
+    #     CziReader,
+    #     LifReader,
+    #     OmeTiffReader,
+    #     TiffReader,
+    #     DefaultReader,
+    # ]
+
+    def __init__(
+        self, image: types.ImageLike, known_dims: Optional[str] = None, **kwargs
+    ):
+        """
+        AICSImage takes microscopy image data types (files or arrays) of varying
+        dimensions ("ZYX", "TCZYX", "CYX") and puts them into a consistent 6D "STCZYX"
+        ordered array. The data and metadata are lazy loaded and can be accessed as
+        needed. Note the dims are assumed to match "STCZYX" from right to left meaning
+        if dimensional data is provided then the dimensions are assigned to be "CZYX", 2
```

Comment:
does "dimensional data" mean `known_dims` ?  This stuff about matching right to left is slightly confusing to me.